automated-orders/packages/blockchain-library/src/clients/evm.ts

```diff
@@ -1,50 +1,89 @@
+import {
+  Config,
+  createConfig,
+  estimateFeesPerGas,
+  estimateGas,
+  getBlockNumber,
+  getBytecode,
+  getGasPrice,
+  getTransactionCount,
+  http,
+  readContract,
+} from '@wagmi/core';
 import { delay } from '@warden-automated-orders/utils';
 import { LRUCache } from 'lru-cache';
-import { AbiEventFragment, AbiFunctionFragment, Bytes, EventLog, FeeData, Transaction, Web3 } from 'web3';
+import {
+  AbiEvent,
+  AbiFunction,
+  Chain,
+  Hex,
+  Log,
+  PublicClient,
+  createPublicClient,
+  createWalletClient,
+  decodeEventLog,
+  encodeFunctionData,
+} from 'viem';
+import { privateKeyToAccount, signTransaction } from 'viem/accounts';
 
 import { IEvmConfiguration } from '../types/evm/configuration.js';
+import { GasFeeData } from '../types/evm/gas.js';
 import { IEventPollingConfiguration } from '../types/evm/pollingConfiguration.js';
 
 export class EvmClient {
-  web3: Web3;
-  signer: string;
+  signer: Hex;
   eventsFromBlocks: Map<string, bigint>;
-  events: LRUCache<string, EventLog>;
+  events: LRUCache<string, Log>;
+  config: Config;
+  client: PublicClient;
+
+  constructor(
+    private configuration: IEvmConfiguration,
+    chain: Chain,
+  ) {
+    this.config = createConfig({
+      chains: [chain],
+      transports: {
+        [chain.id]: http(this.configuration.rpcURL),
+      },
+    });
+
+    this.client = createPublicClient({
+      chain: chain,
+      transport: http(this.configuration.rpcURL),
+    });
 
-  constructor(private configuration: IEvmConfiguration) {
-    this.web3 = new Web3(this.configuration.rpcURL);
     this.eventsFromBlocks = new Map<string, bigint>();
 
     if (this.configuration.eventsCacheSize) {
-      this.events = new LRUCache<string, EventLog>({
+      this.events = new LRUCache<string, Log>({
         max: this.configuration.eventsCacheSize,
       });
```

Comment:
we probably should throw error if this.configuration.eventsCacheSize is undefined

Reply:
Or just make eventsCacheSize mandatory

Reply:
Hm, I see now why it not mandatory - client for non-warden chain not use events cache and not signing anything.

Anyway this values initiated with env-schema and it would throw error if values will be empty